worlds/sa2b/Rules.py

```diff
@@ -1439,7 +1439,9 @@ def set_mission_upgrade_rules_standard(multiworld: MultiWorld, world: World, pla
                  lambda state: state.has(ItemName.rouge_iron_boots, player))
 
         add_rule(multiworld.get_location(LocationName.pyramid_cave_lifebox_5, player),
-                 lambda state: state.has(ItemName.sonic_bounce_bracelet, player))
+                 lambda state: (state.has(ItemName.sonic_light_shoes, player) and
+                                state.has(ItemName.sonic_bounce_bracelet, player) and
+                                state.has(ItemName.sonic_mystic_melody, player)))
```

Comment:
```suggestion
                 lambda state: state.has_all((ItemName.sonic_light_shoes, ItemName.sonic_bounce_bracelet, ItemName.sonic_mystic_melody), player))
```
Recommend using state.has_all, up to you, I'm not gonna argue about it